cachi2/core/package_managers/rpm/main.py

```diff
@@ -15,6 +19,8 @@
 
 
 def fetch_rpm_source(request: Request) -> RequestOutput:
+    """Process the input lockfile, fetch packages and generate SBOM."""
```

Comment:
This doc string should have been part of commit https://github.com/containerbuildsystem/cachi2/pull/502/commits/58bd8eb54b289c98951f7284b57b051fbb17ff00

Reply:
Yes, I did it intentionally. I didn't want to modify this doc string in more commits. I put it in when the doc string was final.
I wasn't prepared for the strict requirement of having clean linters and unittests after every commit despite the commits will be pushed together. I agree original unittests had to be fixed overall, that was my fault.

Reply:
> Yes, I did it intentionally. I didn't want to modify this doc string in more commits. I put it in when the doc string was final.

No need, since we know upfront what the purpose of the method is we can put the doc string in there right away.

> ...despite the commits will be pushed together

What do you mean by pushed together? We don't do squash-merging, so the commits would appear in the history exactly as proposed, so whenever one would run a bisect they'd get false positives on unittests and whatnot stemming from the fact that not every commit passed the unit test suite (let me also state that we have already mistakenly broken this, but in general we're trying to avoid doing that again).